services/scraper/scraper.go

```diff
@@ -4,15 +4,16 @@ import (
 	"fmt"
 	"net/url"
 
-	"go-google-scraper-challenge/helpers"
 	"go-google-scraper-challenge/models"
 
 	"github.com/beego/beego/v2/core/logs"
 	"github.com/gocolly/colly"
-	"github.com/gocolly/colly/queue"
-	"github.com/siddontang/go/num"
 )
 
+type Scraper struct {
+	Result *models.Result
```

Comment:
According to https://github.com/carryall/go-google-scraper-challenge/pull/71#discussion_r643651372.
If you agree with the above comment, I think we can just send the keyword to this service and use `UpdateResultStatus` directly from model (so it should be `models.UpdateResultStatus`)

What do you think?

Reply:
@gutakk Actually I don't think we can pass only the keyword to the service because at least we need to pass a result id (multiple results can have the same keyword) so the service can update the status of the correct result.
I agreed that we should use `UpdateResultStatus` directly from the `models` package but we can still pass the result instance to that function so we don't have to use `models.GetResultById` to get the result instance before updating the status
Also IMO keeping the result instance in the service would be an advantage on updating any result's property and adding other relations, what do you think? 🤔

Reply:
I'm ok with your idea